### PR TITLE
Edit Entities for Create and Modify to use individual builders

### PR DESCRIPTION
**Pull request by jeffreyladd**
# Summary
Updates many of the Create/Modify Entities to use builders.  I also have seperated the builders so they only do Modify or Create.

# Changes Done

1. DiscordGuild Create
    1. Adds missing params on the Create Payload
    2. Simplifies the API to use a builder/Action of a Builder.
    3. Centralizes the validation
2. DiscordGuild Modify
    1. Adds missing params on the modify payload
    2. Simplifies the API to use a builder and reduces the Methods.
    3. Centralizes the validation
3. DiscordChannel Create
    1. Simplifies the API to use a builder and also reduces the amount over Methods and overloads to do one task.
    2. Centralizes the validation
4. DiscordChannel Modify
    1. Simplifies the API to use a builder and also reduces the amount over Methods and overloads to do one task.
    2. Centralizes the validation
5. DiscordRole Create
    1. Simplifies the API to use a builder and also reduces the amount over Methods and overloads to do one task.
    2. Centralizes the validation
6. DiscordRole Modify
    1. Simplifies the API to use a builder and also reduces the amount over Methods and overloads to do one task.
    2. Centralizes the validation
7. DiscordMember Modify
    1. Simplifies the API to use a builder and also reduces the amount over Methods and overloads to do one task.
    2. Centralizes the validation
8. Gluid Membership screening Modify
    1. Simplifies the API to use a builder and also reduces the amount over Methods and overloads to do one task.
    2. Centralizes the validation

# Tests Done
1. Wrote Tests that validate the Guild Create Using the Builder.  Also wrote Tests that validate the Guild Create using an action of a builder.
    1. There is a bug on the discord API side where passing a system_channel_id or afk_channel_id will cause a Server Exception.  This is due to them not handling the mapping correctly.
2. Wrote Tests that validate the Guild Modify Using the Builder.  Also wrote Tests that validate the Guild Modifyusing an action of a builder.
3. Wrote and organized tests that validate everything you can do with the MessageCreate Builder and MessageModifyBuilder.
4. Wrote Tests that validate the Channel Create using the Builder.  Also wrote tests that validate the Channel Create using an action of a builder
5. Wrote Tests that validate the Channel Modify using the Builder.  Also wrote tests that validate the Channel modify using an action of a builder
6. Wrote Tests that validate the Role Create using the Builder.  Also wrote tests that validate the Role Create using an action of a builder
7. Wrote Tests that validate the Role Modify using the Builder.  Also wrote tests that validate the Role modify using an action of a builder
8. Wrote Tests that validate the Member Modify using the Builder.  Also wrote tests that validate the Member modify using an action of a builder
9. Wrote Tests that validate the GuildMembership Modify using the Builder.  Also wrote tests that validate the GuildMembership modify using an action of a builder
    1.  For this i currently dont have the facilities to test this for some reason.  I keep on getting a access denied permission from discord.  Dunno what that is about.

# Items left to do

1. Finish writing tests
2. Update Documentation where we changed the logic
3. Create Developer Documentation when it comes in terms of writing a builder and the format they should follow to make things consistent.

### CODE REVIEW

**Comment by jeffreyladd** [comment]
There currently is a bug when trying to place a user into a voice channel.  @Neuheit can u eyeball if i did something wrong here?  There error i get back is "Target user is not connected to voice."

File: DSharpPlus/Entities/Builders/DiscordMemberModifyBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+            return this;
+        }
+
+        /// <summary>
+        /// Sets if the user should be deafened.
+        /// </summary>
+        /// <param name="deafen">The deafen value.</param>
+        /// <returns></returns>
+        public DiscordMemberModifyBuilder WithDeafned(bool deafen)
```
Comment (Neuheit) [inline DSharpPlus/Entities/Builders/DiscordMemberModifyBuilder.cs:97]: Typo: Should be `WithDeafened`

File: DSharpPlus/Entities/Builders/DiscordMessageBuilder.cs [hunk 2]
```diff
@@ -7,9 +7,9 @@
 namespace DSharpPlus.Entities
 {
     /// <summary>
-    /// Constructs a Message to be sent.
+    /// Represents the message.
```
Comment (Neuheit) [inline DSharpPlus/Entities/Builders/DiscordMessageBuilder.cs:10]: This XML looks incomplete.

File: DSharpPlus/Entities/Builders/DiscordMemberModifyBuilder.cs (hunk 1 above)
Comment (jeffreyladd) [reply DSharpPlus/Entities/Builders/DiscordMemberModifyBuilder.cs]: Fixed

File: DSharpPlus/Entities/Builders/DiscordMessageBuilder.cs (hunk 2 above)
Comment (jeffreyladd) [reply DSharpPlus/Entities/Builders/DiscordMessageBuilder.cs]: fixed

**Comment by Neuheit** [comment]
Putting this on hold for the time being: too complex and need to get the 4.0 release out.

**Comment by Neuheit** [comment]
Closing this due to inactivity as well as for targeting the wrong branch. Please let me know if you wish to continue working on this.

**Pull request closed without merging**